packages/controller/lib/setup/setupInstall.js

```diff
@@ -640,266 +613,235 @@ function Install(options) {
 
         mime = mime || require('mime');
 
-        objects.getObject('system.adapter.' + adapter, (err, doc) => {
-            // Adapter is not installed - install it now
-            if (err || !doc || !doc.common.installedVersion) {
-                return that.installAdapter(adapter, () =>
-                    that.createInstance(adapter, options, callback));
-            }
+        let doc = await objects.getObjectAsync('system.adapter.' + adapter);
+        // Adapter is not installed - install it now
+        if (!doc || !doc.common.installedVersion) {
```

Comment:
original code also ignored the error case and executed the code from this if

Reply:
Added err processing